exotations/task_maps/task_map/include/task_map/JointJerkBackFiniteDiff.h

```diff
@@ -0,0 +1,86 @@
+/*
+ *      Author: Christopher E. Mower
+ *
+ * Copyright (c) 2018, University Of Edinburgh
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of  nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef JOINTJERKBACKFINITEDIFF_H_
+#define JOINTJERKBACKFINITEDIFF_H_
+
+#include <exotica/TaskMap.h>
+#include <task_map/JointJerkBackFiniteDiffInitializer.h>
+
+namespace exotica {
+
+  /**
+   * \brief	Implementation of joint velocity estimation using backward finite differencing task map.
+   */
+
+  class JointJerkBackFiniteDiff : public TaskMap, public Instantiable<JointJerkBackFiniteDiffInitializer>
+  {
+
+  public:
+
+    // Default constructor
+    JointJerkBackFiniteDiff();
+
+    // Default destructor
+    ~JointJerkBackFiniteDiff();
+
+    virtual void Instantiate(JointJerkBackFiniteDiffInitializer& init);
+
+    void Initialize();
+
+    void setPrevJointState(Eigen::VectorXd joint_state);
```

Comment:
Consider using ``Eigen::VectorXdRefConst`` (or similar - avoids copy).

Reply:
Covered in latest commit.